src/server.cpp

```diff
@@ -388,6 +389,12 @@ void Server::init()
 
 	m_script = new ServerScripting(this);
 
+	// Initialize Environment
+	m_env = new ServerEnvironment(servermap, m_script, this, m_path_world);
```

Comment:
This is a pretty significant change, I don't think the environment should be initialized before mods are loaded.
If you need `m_inventory_mgr` to be created at this point, it should rather have a method to set the `m_env` when its actually created.
( see also my comment about the `if (!m_env)`)

Reply:
environment is just a constructor and doesn't run or register anything, it just open the world.mt configuration currently. Also mods rely on environment, they must load after, i can put a setter but it's more error prone in terms of init that anythings, adding more spaghettis in our init code.

Reply:
> Also mods rely on environment, they must load after

No they do not, mods have always been loaded before the environment was created.

The initialization order should not just be changed because it's convenient. Also, if this is done warnings like this will not work anymore:
https://github.com/minetest/minetest/blob/cb159f8d8af4556391db8a6875657625733eeb11/src/script/lua_api/l_env.cpp#L643-L648

Reply:
understood. I restored the old behaviour on detached inventory creation.